fittings.gemspec

```diff
@@ -20,6 +20,7 @@ Gem::Specification.new do |s|
   s.executables   = s.files.grep(%r{^exe/}) { |f| File.basename(f) }
   s.require_paths = ["lib"]
 
+  s.add_dependency "hashie"
```

Comment:
I'm hesitant to add this dependency to every app that uses fittings. Is this bug something that can be fixed without it?

Reply:
It can be, yes. We would have to either globally monkey patch `Hash` or recursively wrap each object in an enhanced hash, which is essentially what `Hashie::Mash` is doing for us. The tradeoffs with those approaches being that we avoid a new dependency, but incur supporting similar complex logic on our own (the recursive logic) or extend our monkey patching of a core Rails class. I personally think we should avoid monkey patching Ruby classes due to the risk that can bring. Too much depends on them for us to be tweaking them on our own.

I can see the arguments for the recursive option and this option. I lean toward this approach because it minimizes our future support needs for this gem, but I'm definitely open on the issue.

Reply:
Just a data point: per appetizer, it looks like every app that uses `fittings` already uses `hashie` as well. There may be a flaw in my method, posting here for review/scrutiny:

```
appetizer -g hashie | jq -r keys[] > hashie
appetizer -g fittings | jq -r keys[] > fittings
comm -23 fittings hashie
# no output, meaning no lines unique to "fittings"
```

Reply:
Ok. I don't want to block progress here. I've seen dependencies prevent upgrades to Rails and Ruby. And every dependency is an extra download for everyone.

But if everyone already needs it... this is probably fine.

Reply:
Sorry to be giving late feedback—I hinted at this in slack, but api_client gem will always pull in `hashie` as a dependency ([for backwards-compatibility with code that was written with the version that used hashie](https://github.com/stitchfix/api_client/blob/main/lib/stitch_fix/api_client/faraday_middleware/response_wrapper.rb#L18-L22)), but it's very common to [configure subclasses](https://github.com/stitchfix/question_set_service_client/blob/main/lib/stitch_fix/question_set_service_client/v1_client.rb#L9) or [other uses](https://github.com/stitchfix/dasher/blob/main/app/services/client_context_service.rb#L38) to use plain hashes instead, so teams are able to opt-in to Hashie.